src/coreclr/jit/compiler.cpp

```diff
@@ -5133,6 +5133,30 @@ void Compiler::compCompile(void** methodCodePtr, uint32_t* methodCodeSize, JitFl
     StackLevelSetter stackLevelSetter(this);
     stackLevelSetter.Run();
 
+    // If optimizations are enabled, we may remove throw blocks when setting stack levels.
+    // This can move a BBJ_COND's true target to its next block.
+    // In such cases, reverse the condition so we can remove a branch.
+    if (opts.OptimizationEnabled())
+    {
+        for (BasicBlock* const block : Blocks())
```

Comment:
This loop seems to be responsible for the TP regressions on Windows x86. Oddly enough, I don't see this putting a dent in the TP improvements on x64, and the x86 collections for which TP regressed also had pretty big size improvements. Should I save this for a follow-up PR so we can independently observe its cost/benefits?